### ISSUE TITLE
A Craft Discount redeemed through GiftVoucher input generates error

### ISSUE THREAD

**Issue by TomDeSmet** [verbb/gift-voucher#96]
### Describe the bug

When using the GiftVoucher input field (copied from the docs) in our cart template, and adding a Craft Discount code (so not a Gift Voucher) we get the following error:

`Cannot assign null to property verbb\giftvoucher\events\MatchCodeEvent::$code of type verbb\giftvoucher\elements\Code`

Tracing the bug and looking at the code didn't give me a clear view of what's wrong as the code seems not to have changed from GiftVoucher 2.7.0 where it did work.

### Steps to reproduce

1. Create a Craft discount
2. Add it on a cart template to the GiftVoucher redeem input form.
3. Apply the discount

### Craft CMS version

4.2.5.2

### Plugin version

3.0.0

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get the fix early, change your `verbb/gift-voucher` requirement in `composer.json` to:

```
"require": {
  "verbb/gift-voucher": "dev-craft-4 as 3.0.0",
  "...": "..."
}
```

Then run `composer update`.

**Comment by tomdeleu** [comment]
Probably related. But when adding a Craft coupon I'm getting another error since Craft discounts can have multiple coupon codes now:
`"Getting unknown property: craft\\commerce\\models\\Discount::code",`

Edit: It only seems to occur when I combine a voucher and a coupon code, and if 'stopProcessing' is true. Just a native Craft coupon seems to work fine or setting 'stopProcessing' works fine too.

**Comment by engram-design** [comment]
Fixed in [3.0.1](https://github.com/verbb/gift-voucher/releases/tag/3.0.1)
